zuul-tests.d/magic-jobs.yaml

```diff
@@ -0,0 +1,20 @@
+- job:
```

Comment:
Please rename the file itself (i.e. containerized-jobs). It has nothing to do with magic, but a workaround for things you do not like. Having proper jobs gives future flexibility to run jobs themselves in containers instead of VMs (on the roadmap), while docker in docker is not something incredibly practical. On the other hand this type of job, while making your personal life easier, is absolutely not reusable by other projects.

Reply:
Ok, will rename. I really don't get why is it not reusable.
The main point of using docker here is to provide the possibility to select the required base image without creating a new job for that

Reply:
It does not give possibility to put build artifacts into Zuul build results (see https://zuul.opendev.org/t/zuul/build/08d7603460904d8c89ddab3cbc5a1a6d/artifacts, https://zuul.opendev.org/t/zuul/build/e2c72feb9d7a42e8a424c309859894d6/artifacts for examples), publish build artifacts later by i.e. even chaining jobs, it doesn't allow centralizing how particular project types should be build/structured.
Don't get me wrong - the job on it's own is reusable, but it doesn't help other projects to follow some/same guidelines (i.e. on a central level say from now on we use version yarn x.y for all js projects) and neglects idea of Zuul.